src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = century
+    ? people.filter((p) => p.sex === 'm' && Math.ceil(p.died / 100) === century)
+    : people.filter((p) => p.sex === 'm');
+
+  let sum = 0;
+
+  men.forEach((m) => sum += m.died - m.born);
```

Comment:
Refactor this line:

There isn't a big problem in using a .forEach, generally speaking. (It is better than using a for-of or for-in loop), however, the task asks specifically to avoid using it.
In this case here, as all you want is a single value, **sum**, you can use a **.reduce** to assign it. Here is an example:
![image](https://github.com/pedro-ruas/mate-js_average-ages_test-task/assets/54222238/4e7b5c66-451f-42dd-a7e3-89d30df33fb9)
A bonus is that, using this reduce method, you can remove the **let sum** declaration.